File: src/main/scala/org/lolhens/minechanics/common/gui/GuiHandler.scala
package org.lolhens.minechanics.common.gui

import net.minecraft.entity.player.EntityPlayer
import net.minecraft.world.World
import net.minecraftforge.fml.common.network.IGuiHandler

/**
 * Created by LolHens on 19.09.2014.
 */
class GuiHandler extends IGuiHandler {
  override def getServerGuiElement(ID: Int, player: EntityPlayer, world: World, x: Int, y: Int, z: Int): AnyRef = ???

  override def getClientGuiElement(ID: Int, player: EntityPlayer, world: World, x: Int, y: Int, z: Int): AnyRef = ???
}

File: src/main/java/org/lolhens/minechanics/core/gui/GuiHandler.java
package org.lolhens.minechanics.core.gui;

import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.world.World;
import cpw.mods.fml.common.network.IGuiHandler;

public class GuiHandler implements IGuiHandler {
    @Override
    public Object getServerGuiElement(int ID, EntityPlayer player, World world, int x, int y, int z) {
        return Gui.fromID(ID).getNewContainerInstance(player, world, x, y, z);
    }

    @Override
    public Object getClientGuiElement(int ID, EntityPlayer player, World world, int x, int y, int z) {
        return Gui.fromID(ID).getNewGuiContainerInstance(player, world, x, y, z);
    }
}
